journal/profiles.cpp

```diff
@@ -49,11 +49,25 @@ T DeserializePointer(Player::PointerMap const& pointer_map,
   }
 }
 
+char16_t const* DeserializeUtf16(std::string const& serialized) {
+  // The result is char16_t-null-terminated because of the way the serialized
+  // string was built in SerializeUtf16.
+  return reinterpret_cast<char16_t const*>(serialized.c_str());
```

Comment:
Accessing a value through `*DeserializeUtf16` is explicitly undefined, see [N4659](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/n4659.pdf) 6.10(8).

Reply:
Done.